libfwupd/fwupd-client-sync.c

```diff
@@ -2595,6 +2595,70 @@ fwupd_client_upload_bytes(FwupdClient *self,
 	return g_steal_pointer(&helper->bytes);
 }
 
+static void
+fwupd_client_upload_report_cb(GObject *source, GAsyncResult *res, gpointer user_data)
+{
+	FwupdClientHelper *helper = (FwupdClientHelper *)user_data;
+	helper->str = fwupd_client_upload_report_finish(FWUPD_CLIENT(source), res, &helper->error);
+	g_main_loop_quit(helper->loop);
+}
+
+/**
+ * fwupd_client_upload_report:
+ * @self: a #FwupdClient
+ * @url: (not nullable): the remote URL
+ * @payload: (not nullable): payload string
+ * @signature: (nullable): signature string
+ * @flags: download flags, e.g. %FWUPD_CLIENT_DOWNLOAD_FLAG_NONE
+ * @cancellable: (nullable): optional #GCancellable
+ * @error: (nullable): optional return location for an error
+ *
+ * Uploads a report to a remote server. The [method@Client.set_user_agent] function
+ * should be called before this method is used.
+ *
+ * Returns: (transfer full): a URI (perhaps an empty string), or %NULL for error
+ *
+ * Since: 1.9.20
```

Comment:
BTW I'm totally fine if this is for 2.0.0 only and 2.0.0 bumps up min glib deps

Reply:
No worries; i can use an older API.